src/common/types/vector.cpp

```diff
@@ -1068,6 +1073,147 @@ void ListVector::Initialize(Vector &vec) {
 		ListVector::SetEntry(vec, move(vec_child));
 	}
 }
+
+ValidityMask &Vector::GetValidity(Vector &v) {
+	switch (v.GetVectorType()) {
+	case VectorType::DICTIONARY_VECTOR:
+		return GetValidity(DictionaryVector::Child(v));
+	case VectorType::FLAT_VECTOR:
+		return FlatVector::Validity(v);
+	case VectorType::CONSTANT_VECTOR:
+		return ConstantVector::Validity(v);
+	default:
+		throw NotImplementedException("FIXME: cannot deserialize vector with this vectortype");
+	}
+}
+
+template <class T>
+void Search(Vector &list, T key, vector<idx_t> &offsets, bool is_key_null) {
```

Comment:
`Search` is rather generic, which might cause problems in the amalgamated version. Perhaps rename to `TemplatedSearchInMap`?